### ISSUE TITLE
Not seeing connection close() entries?

### ISSUE THREAD

**Issue by pmb5** [MiniProfiler/dotnet#399]
My understanding of TrackConnectionOpenClose, is that i should be able to see "Connection Closed()" custom timings when Ado.net closes a connection, but even in the most basic asp.net (webforms) hello world app.  I'm only seeing the "Connection Opened()"

I took a quick look in the source, and i can only find code for EF connections closing, so is this not supported for the ProfiledDbConnection wrapping SQlConnection? I was expecting to see "closing timings", like in this issue, for ADO.net too.

https://github.com/MiniProfiler/dotnet/issues/213

Thanks in Advance.

**Comment by NickCraver** [comment]
How are you opening the connection? Can you show some code? Everything in MiniProfiler depends on using the `ProfiledDbConnection`, ultimately.

**Comment by pmb5** [comment]
Hey Nick,
This is my sample code, just thrown in the default.aspx Page_Load of a webforms app

```cs
        protected void Page_Load(object sender, EventArgs e)
        {
            using (MiniProfiler.Current.CustomTiming("where is my close","?"))
            using (var connection = new ProfiledDbConnection(new SqlConnection("removed to protect-Squirrels"), MiniProfiler.Current))
            {
                var sqlCommand = new SqlCommand("select * from MiniProfilerTimings");
                var cmd = new ProfiledDbCommand(sqlCommand, connection, MiniProfiler.Current);

                connection.Open();
                var reader = cmd.ExecuteReader();
                while (reader.HasRows && reader.Read())
                {
                    Response.Write(reader.GetValue(0));
                }

                //Lets try some explicit closing
                connection.Close();
                connection.WrappedConnection.Close();

                using (MiniProfiler.Current.Step($"'{connection.State}' - The connection state"))
                {
                    Thread.Sleep(100);
                }
            }
            using (MiniProfiler.Current.Step("This is weird"))
            {
                Thread.Sleep(600);
            }
        }
```

This is what my profile results look like, i was expecting to see a custom timing with the  command "Connection Close()"

https://user-images.githubusercontent.com/2781908/30055802-7677b162-9228-11e7-8ab4-557736250192.png
![image](https://user-images.githubusercontent.com/16664046/60805523-43f83c00-a178-11e9-940a-8d77f88df3d8.png)

**Comment by NickCraver** [comment]
Alrighty I finally got time to dig into this today, the story is a bit more complicated which is _probably_ why it was left up in the air.

I can override the `.Close()` method and log it - your example will work. But, it won't work for all scenarios. I wrote up the details in the commit here: https://github.com/MiniProfiler/dotnet/commit/a7322be1d97be0720832ea9667105c0729d9343d

It's a preview release, but the non-`netcoreapp3.0` bits are solid. It'll be up on MyGet shortly :)
